app/home/page.tsx

```diff
@@ -1,6 +1,16 @@
+'use client';
 import Link from "next/link";
+import { useEffect } from "react";
 
 export default function HomePage() {
+  useEffect(() => {
```

Comment:
shouldn't this be in providers.tsx or layout.tsx? Since this worker will not be registered if the user never visits the HomePage